flag all modules used for DLL (#3454)
webpack/webpack.js.org: src/content/plugins/dll-plugin.md

- `context` (optional): context of requests in the manifest file (defaults to the webpack context.)
- `name`: name of the exposed dll function ([TemplatePaths](https://github.com/webpack/webpack/blob/master/lib/TemplatedPathPlugin.js): `[hash]` & `[name]` )
- `path`: __absolute path__ to the manifest json file (output)

> - `entryOnly`: if `true`, only entry points will be exposed (defaults to `true`)

Reviewer: ```suggestion
- `entryOnly (boolean = true)`: if `true`, only entry points will be exposed
```

- `type`: type of the dll bundle

```javascript
new webpack.DllPlugin(options);
```

> Using the DllPlugin without `entryOnly: true` will disable Tree Shaking in the DLL as all exports might be used.

Reviewer: ```suggestion
W> We recommend using DllPlugin only with `entryOnly: true`, otherwise tree shaking in the DLL won't work as all the exports might be used.
```

Creates a `manifest.json` which is written to the given `path`. It contains mappings from require and import requests to module ids. It is used by the `DllReferencePlugin`.

Combine this plugin with [`output.library`](/configuration/output/#outputlibrary) option to expose (aka, put into the global scope) the dll function.

> W> We recommend using only `entryOnly: true`

Reviewer: ```suggestion
```

new webpack.DllPlugin(options);
```

W> We recommend using DllPlugin only with `entryOnly: true`, otherwise tree shaking in the DLL won't work as all the exports might be used.

Creates a `manifest.json` which is written to the given `path`. It contains mappings from require and import requests to module ids. It is used by the `DllReferencePlugin`.

Combine this plugin with [`output.library`](/configuration/output/#outputlibrary) option to expose (aka, put into the global scope) the dll function.

> 

Reviewer: ```suggestion
```
